test/unittests/test_analyse_xml_data/test_extract_links_from_XML.py

```diff
@@ -0,0 +1,142 @@
+# pylint: disable=missing-class-docstring,missing-function-docstring,protected-access
+
+import pytest
+from lxml import etree
+from pytest_unordered import unordered
+
+from dsp_tools.analyse_xml_data.extract_links_from_XML import (
+    _create_classes_from_single_resource,
+    _extract_ids_from_one_resptr_prop,
+    _extract_ids_from_one_text_value,
+    _extract_ids_from_text_prop,
+    _get_all_links_from_one_resource,
+)
+
+
+def test_create_classes_from_single_resource() -> None:
+    test_ele = etree.fromstring(
+        '<resource xmlns="https://dasch.swiss/schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+        'label="res_A_19" restype=":TestThing" id="res_A_19" permissions="res-default"><resptr-prop '
+        'name=":hasResource1"><resptr permissions="prop-default">res_B_19</resptr><resptr '
+        'permissions="prop-default">res_C_19</resptr></resptr-prop><text-prop name=":hasRichtext"><text '
+        'permissions="prop-default" encoding="xml"><a class="salsah-link" href="IRI:res_B_19:IRI">res_B_19</a><a '
+        'class="salsah-link" href="IRI:res_C_19:IRI">res_C_19</a></text></text-prop></resource>'
+    )
+    res_resptr_links, res_xml_links, res_all_used_ids = _create_classes_from_single_resource(test_ele)
+    #########
+    res_B_19 = [obj.object_id for obj in res_resptr_links]  # type: ignore[union-attr]
+    assert "res_B_19" in res_B_19
+    res_C_19 = [obj.object_id for obj in res_resptr_links]  # type: ignore[union-attr]
+    assert "res_C_19" in res_C_19
+    #########
+    expected_ids = ["res_B_19", "res_C_19", "res_B_19", "res_C_19", "res_A_19"]
+    assert expected_ids == unordered(res_all_used_ids)
+    ########
+    expected_xml = {"subject_id": "res_A_19", "object_link_ids": {"res_B_19", "res_C_19"}}
+    res_xml_dict = res_xml_links[0].__dict__  # type: ignore[index]
+    for k, v in expected_xml.items():
+        assert v == res_xml_dict[k]
+
+
+def test_get_all_links_from_one_resource() -> None:
+    test_ele = etree.fromstring(
+        '<resource xmlns="https://dasch.swiss/schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+        'label="res_A_11" restype=":TestThing" id="res_A_11" permissions="res-default"><text-prop '
+        'name=":hasRichtext"><text permissions="prop-default" encoding="xml"><a class="salsah-link" '
+        'href="IRI:res_B_11:IRI">res_B_11</a></text></text-prop><resptr-prop name=":hasResource1"><resptr '
+        'permissions="prop-default">res_B_11</resptr></resptr-prop></resource>'
+    )
+    res_resptr, res_xml = _get_all_links_from_one_resource(test_ele)
+    expected_resptr, expected_xml = ["res_B_11"], [{"res_B_11"}]
+    assert expected_resptr == res_resptr
+    assert expected_xml == unordered(res_xml)
```

Comment:
This can be done easier:

```suggestion
    assert res_resptr == ["res_B_11"]
    assert unordered(res_xml) == [{"res_B_11"}]
```